### PR TITLE
[MM-52002] Improve Calls settings documentation

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Improving config settings documentation for Calls, adding some notes, a missing setting and an example.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-52002

### CODE REVIEW

File: source/configure/plugins-configuration-settings.rst [hunk 1]
```diff
@@ -505,10 +505,34 @@ Enable plugin
 | - **false**: Disables the Calls plugin on your Mattermost workspace. | - ``config.json`` setting:                |
 +----------------------------------------------------------------------+-------------------------------------------+
 
+.. config:setting:: plugins-callsrtcserveraddress
+  :displayname: RTC server port (Plugins - Calls)
+  :systemconsole: Plugins > Calls
+  :configjson:
```
Comment (cwarnermm) [inline source/configure/plugins-configuration-settings.rst:511]: If this config setting doesn't have an equivalent ``config.json`` setting, please populate the ``config.json`` and ``environment`` fields here as ``N/A``

File: source/configure/plugins-configuration-settings.rst [hunk 2]
```diff
@@ -505,10 +505,34 @@ Enable plugin
...
+
+RTC server address
+~~~~~~~~~~~~~~~~~~
+
+.. include:: ../_static/badges/selfhosted-only.rst
+  :start-after: :nosearch:
+
++-----------------------------------------------------------------------------------------------------------------------+-------------------------------------------+
+| This setting controls the IP address listened on by the RTC server. All Calls traffic will be served through this IP. | - System Config path: **Plugins > Calls** |
+|                                                                                                                       | - ``config.json`` setting:                |
```
Comment (cwarnermm) [inline source/configure/plugins-configuration-settings.rst:523]: If this config setting doesn't have an equivalent ``config.json`` setting, please populate the ``config.json`` field in this table as ``N/A``, and include the ``Environment variable`` line to the table with a value of ``N/A``.

File: source/configure/plugins-configuration-settings.rst [hunk 3]
```diff
@@ -505,10 +505,34 @@ Enable plugin
...
+  :description: The IP address used by the RTC server to listen on. By default the service listens on all the available interfaces.
+
+RTC server address
+~~~~~~~~~~~~~~~~~~
+
+.. include:: ../_static/badges/selfhosted-only.rst
+  :start-after: :nosearch:
+
++-----------------------------------------------------------------------------------------------------------------------+-------------------------------------------+
+| This setting controls the IP address listened on by the RTC server. All Calls traffic will be served through this IP. | - System Config path: **Plugins > Calls** |
```
Comment (cwarnermm) [inline source/configure/plugins-configuration-settings.rst:522]: consider instead: "This setting controls the IP address the RTC server listens to."

**Review by cwarnermm — COMMENTED** [review]
Thanks, @streamer45! I've provided feedback inline.

**Review by cpoile — APPROVED** [review]

File: source/configure/plugins-configuration-settings.rst [hunk 4]
```diff
@@ -491,25 +491,49 @@ Access the following configuration settings in the System Console by going to **
...
+| Changing this setting requires a plugin restart to take effect.                                                       | - Environment variable: N/A                                                                               |
+|                                                                                                                       |                                                                                                           |
+| If left unset (default value) the service will listen on all the available interfaces.                                |                                                                                                           |
++-----------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Note**: This setting is only applicable when not running calls through the standalone ``rtcd`` service.             |                                                                                                           |
```
Comment (cwarnermm) [inline source/configure/plugins-configuration-settings.rst:529]:
```suggestion
| **Note**: This setting is only applicable when not running calls through the standalone ``rtcd`` service.                                                                                                                         |
```

File: source/configure/plugins-configuration-settings.rst [hunk 5]
```diff
@@ -518,19 +542,22 @@ RTC server port
...
+| Changing this setting requires a plugin restart to take effect.                                                       |                                                                                                        |
+|                                                                                                                       |                                                                                                        |
+| Default is **8443**.                                                                                                  |                                                                                                        |
++-----------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------+
+| **Note**: This setting is only applicable when not running calls through the standalone ``rtcd`` service.             |                                                                                                        |
```
Comment (cwarnermm) [inline source/configure/plugins-configuration-settings.rst:553]:
```suggestion
| **Note**: This setting is only applicable when not running calls through the standalone ``rtcd`` service.                                                                                                                      |
```

File: source/configure/plugins-configuration-settings.rst [hunk 6]
```diff
@@ -629,15 +659,16 @@ ICE servers configurations
...
+|                                                                                                 | - ``config.json`` setting: ``PluginSettings.Plugins.com.mattermost.calls.iceserversconfigs``                |
+|                                                                                                 | - Environment variable: N/A                                                                                 |
+| This is an optional field. Changing this setting may require a plugin restart to take effect.   |                                                                                                             |
+|                                                                                                 |                                                                                                             |
+| Default is **[{"urls": ["stun:stun.global.calls.mattermost.com:3478"]}]**                       |                                                                                                             |
```
Comment (cwarnermm) [inline source/configure/plugins-configuration-settings.rst:668]:
```suggestion
| Default is ``[{"urls": ["stun:stun.global.calls.mattermost.com:3478"]}]``                       |                                                                                                             |
```

File: source/configure/plugins-configuration-settings.rst [hunk 7]
```diff
@@ -657,12 +688,23 @@ ICE servers configurations
    ]
  
  
+**Example (Using generated TURN credentials)**
```
Comment (cwarnermm) [inline source/configure/plugins-configuration-settings.rst:691]: please indent this example as well for consistency.

**Review by cwarnermm — APPROVED** [review]
I've provided minor formatting feedback inline. Approving.

**Comment by streamer45** [comment]
Will merge this once ready as we released the new plugin version so it makes sense to update documentation.

**Pull request merged**